pylsp/python_lsp.py

```diff
@@ -91,32 +91,114 @@ def start_io_lang_server(rfile, wfile, check_parent_process, handler_class):
     server.start()
 
 
+def start_ws_lang_server(bind_addr, port, check_parent_process, handler_class):
+    if not issubclass(handler_class, PythonLSPServer):
+        raise ValueError('Handler class must be an instance of PythonLSPServer')
+
+    # pylint: disable=import-outside-toplevel
+
+    # imports needed only for websocket based server
+    from autobahn.twisted.websocket import WebSocketServerProtocol, WebSocketServerFactory
+    from twisted.internet import reactor
```

Comment:
You could wrap these with try-except and re-raise with instructions for installing websocket requirements, something like "Please run `pip install 'python-lsp-server[autobahn-twisted]` or equivalent".